### ISSUE TITLE
Tests fail during build with 'java.io.IOException: Network is unreachable'

### ISSUE THREAD

**Issue by Wedjaa** [redhat-italy/hacep#33]
On linux the build process will fail during the tests when there are interfaces plumbed up with IPv6.

Build will succeed if run with:

``
mvn -P community -Djava.net.preferIPv4Stack=true clean install
``

**Comment by fmarinelli** [comment]
Can you please attach the output from your build with IPv6 networking issue?

**Comment by Wedjaa** [comment]
Building with


``
mvn -P community clean install
``

I get the following build error:

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running it.redhat.hacep.camel.ResequencerTest
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Tests run: 2, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 1.399 sec - in it.redhat.hacep.camel.ResequencerTest
Running it.redhat.hacep.compressor.CompressorTest
Compression time: 1
Compression time: 1
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec - in it.redhat.hacep.compressor.CompressorTest
Running it.redhat.hacep.cluster.ClusterTest
Start test cluster size
Start node with owners(2)
Aug 04, 2016 3:42:18 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport start
INFO: ISPN000078: Starting JGroups channel HACEP

-------------------------------------------------------------------
GMS: address=perkele-59736, cluster=HACEP, physical address=127.0.0.1:7800
-------------------------------------------------------------------
Aug 04, 2016 3:42:18 PM org.jgroups.protocols.MPING sendMcastDiscoveryRequest
SEVERE: JGRP000200: failed sending discovery request
java.io.IOException: Network is unreachable
	at java.net.PlainDatagramSocketImpl.send(Native Method)
	at java.net.DatagramSocket.send(DatagramSocket.java:693)
	at org.jgroups.protocols.MPING.sendMcastDiscoveryRequest(MPING.java:295)
	at org.jgroups.protocols.PING.sendDiscoveryRequest(PING.java:62)
	at org.jgroups.protocols.PING.findMembers(PING.java:32)
	at org.jgroups.protocols.Discovery.findMembers(Discovery.java:244)
	at org.jgroups.protocols.Discovery.down(Discovery.java:385)
	at org.jgroups.protocols.MERGE3.down(MERGE3.java:254)
	at org.jgroups.protocols.FD_SOCK.down(FD_SOCK.java:371)
	at org.jgroups.protocols.FD.down(FD.java:318)
	at org.jgroups.protocols.VERIFY_SUSPECT.down(VERIFY_SUSPECT.java:92)
	at org.jgroups.protocols.BARRIER.down(BARRIER.java:132)
	at org.jgroups.protocols.pbcast.NAKACK2.down(NAKACK2.java:589)
	at org.jgroups.protocols.UNICAST3.down(UNICAST3.java:669)
	at org.jgroups.protocols.pbcast.STABLE.down(STABLE.java:347)
	at org.jgroups.protocols.pbcast.ClientGmsImpl.joinInternal(ClientGmsImpl.java:76)
	at org.jgroups.protocols.pbcast.ClientGmsImpl.join(ClientGmsImpl.java:41)
	at org.jgroups.protocols.pbcast.GMS.down(GMS.java:1085)
	at org.jgroups.protocols.FlowControl.down(FlowControl.java:351)
	at org.jgroups.protocols.FRAG2.down(FRAG2.java:136)
	at org.jgroups.protocols.pbcast.STATE_TRANSFER.down(STATE_TRANSFER.java:202)
	at org.jgroups.stack.ProtocolStack.down(ProtocolStack.java:1039)
	at org.jgroups.JChannel.down(JChannel.java:790)
	at org.jgroups.JChannel._connect(JChannel.java:564)
	at org.jgroups.JChannel.connect(JChannel.java:294)
	at org.jgroups.JChannel.connect(JChannel.java:279)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.startJGroupsChannelIfNeeded(JGroupsTransport.java:200)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.start(JGroupsTransport.java:191)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.infinispan.commons.util.ReflectionUtil.invokeAccessibly(ReflectionUtil.java:168)
	at org.infinispan.factories.AbstractComponentRegistry$PrioritizedMethod.invoke(AbstractComponentRegistry.java:859)
	at org.infinispan.factories.AbstractComponentRegistry.invokeStartMethods(AbstractComponentRegistry.java:628)
	at org.infinispan.factories.AbstractComponentRegistry.internalStart(AbstractComponentRegistry.java:617)
	at org.infinispan.factories.AbstractComponentRegistry.start(AbstractComponentRegistry.java:542)
	at org.infinispan.factories.GlobalComponentRegistry.start(GlobalComponentRegistry.java:236)
	at org.infinispan.manager.DefaultCacheManager.start(DefaultCacheManager.java:663)
	at org.infinispan.manager.DefaultCacheManager.<init>(DefaultCacheManager.java:232)
	at it.redhat.hacep.cluster.AbstractClusterTest.clusteredCacheManager(AbstractClusterTest.java:79)
	at it.redhat.hacep.cluster.AbstractClusterTest.startDistSyncNode(AbstractClusterTest.java:53)
	at it.redhat.hacep.cluster.ClusterTest.testClusterSize(ClusterTest.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.JUnit45AndHigherRunnerImpl.run(JUnit45AndHigherRunnerImpl.java:37)
	at org.mockito.runners.MockitoJUnitRunner.run(MockitoJUnitRunner.java:62)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:367)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:274)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:161)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)

Aug 04, 2016 3:42:20 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport viewAccepted
INFO: ISPN000094: Received new cluster view for channel HACEP: [perkele-59736|0] (1) [perkele-59736]
Aug 04, 2016 3:42:21 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport startJGroupsChannelIfNeeded
INFO: ISPN000079: Channel HACEP local address is perkele-59736, physical addresses are [127.0.0.1:7800]
Aug 04, 2016 3:42:21 PM org.infinispan.factories.GlobalComponentRegistry start
INFO: ISPN000128: Infinispan version: Infinispan 'Chakra' 8.2.2.Final
Aug 04, 2016 3:42:21 PM org.infinispan.transaction.lookup.GenericTransactionManagerLookup useDummyTM
WARN: ISPN000104: Falling back to DummyTransactionManager from Infinispan
Start node with owners(2)
Aug 04, 2016 3:42:21 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport start
INFO: ISPN000078: Starting JGroups channel HACEP
....................
....................
Similar Output Omitted
....................
....................
Aug 04, 2016 3:42:44 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport viewAccepted
INFO: ISPN000094: Received new cluster view for channel HACEP: [perkele-16510|0] (1) [perkele-16510]
Aug 04, 2016 3:42:44 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport startJGroupsChannelIfNeeded
INFO: ISPN000079: Channel HACEP local address is perkele-16510, physical addresses are [127.0.0.1:7801]
Stopping cache manager: [perkele-18282]
Stopping cache manager: [perkele-16510]
Aug 04, 2016 3:42:44 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport stop
INFO: ISPN000080: Disconnecting JGroups channel HACEP
Aug 04, 2016 3:42:44 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport stop
INFO: ISPN000082: Stopping the RpcDispatcher for channel HACEP
Aug 04, 2016 3:42:44 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport stop
INFO: ISPN000080: Disconnecting JGroups channel HACEP
Aug 04, 2016 3:42:44 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport stop
INFO: ISPN000082: Stopping the RpcDispatcher for channel HACEP
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 4.12 sec <<< FAILURE! - in it.redhat.hacep.rules.TestModifiedRules
testNonEmptyHASession(it.redhat.hacep.rules.TestModifiedRules)  Time elapsed: 4.119 sec  <<< FAILURE!
java.lang.AssertionError
	at it.redhat.hacep.rules.TestModifiedRules.testNonEmptyHASession(TestModifiedRules.java:75)


Results :

Failed tests: 
  ClusterTest.testClusterSize:64 expected:<2> but was:<1>
  ClusterTest.testEmptyHASession:87
  ClusterTest.testHASessionAddNode:246
  ClusterTest.testHASessionWithMaxBuffer:191
  ClusterTest.testNonEmptyHASession:134
  TestModifiedRules.testNonEmptyHASession:75

Tests run: 10, Failures: 6, Errors: 0, Skipped: 1

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] HACEP Parent ....................................... SUCCESS [  0.184 s]
[INFO] HACEP Commons Test ................................. SUCCESS [  0.945 s]
[INFO] HACEP Core Model ................................... SUCCESS [  0.079 s]
[INFO] HACEP Core ......................................... FAILURE [ 29.093 s]
[INFO] HACEP Examples ..................................... SKIPPED
[INFO] HACEP Example Model ................................ SKIPPED
[INFO] HACEP Example Rules ................................ SKIPPED
[INFO] HACEP Performance Client ........................... SKIPPED
[INFO] HACEP Playground ................................... SKIPPED
[INFO] HACEP EAP Playground ............................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 31.040 s
[INFO] Finished at: 2016-08-04T15:42:44-05:00
[INFO] Final Memory: 49M/623M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project hacep-core: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/mrwho/Projects/Personal/es_hacep/hacep/hacep-core/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :hacep-core
```

Using:

```
mvn -P community -Djava.net.preferIPv4Stack=true  clean install
```

I get the expected results:

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] HACEP Parent ....................................... SUCCESS [  0.228 s]
[INFO] HACEP Commons Test ................................. SUCCESS [  0.881 s]
[INFO] HACEP Core Model ................................... SUCCESS [  0.054 s]
[INFO] HACEP Core ......................................... SUCCESS [ 21.886 s]
[INFO] HACEP Examples ..................................... SUCCESS [  0.002 s]
[INFO] HACEP Example Model ................................ SUCCESS [  0.085 s]
[INFO] HACEP Example Rules ................................ SUCCESS [  2.109 s]
[INFO] HACEP Performance Client ........................... SUCCESS [  0.416 s]
[INFO] HACEP Playground ................................... SUCCESS [  0.220 s]
[INFO] HACEP EAP Playground ............................... SUCCESS [  1.204 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 27.787 s
[INFO] Finished at: 2016-08-04T15:46:40-05:00
[INFO] Final Memory: 65M/795M
[INFO] ------------------------------------------------------------------------
```

**Comment by ugol** [comment]
Yup, IPv6/JDK is a known JGroups issue.

**Comment by Wedjaa** [comment]
I think - as you do with the playground node starting examples - you could add the '-Djava.net.preferIPv4Stack=true' flag in the instructions.

**Comment by Wedjaa** [comment]
Closed by commit bafce46
